src/app/app-auth.service.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { TestBed, inject } from '@angular/core/testing';
+
+import { AppAuthService } from './app-auth.service';
+import {AppModule} from './app.module';
+import {APP_BASE_HREF} from '@angular/common';
+
+describe('AppAuthService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        AppAuthService,
+        {provide: APP_BASE_HREF, useValue: 'http://api-staging.dd-decaf.eu/iam'},
+      ],
+      imports: [AppModule],
+    });
+  });
+
+ it('should trust any URL by default', inject([AppAuthService], (service: AppAuthService) => {
+    expect(service.isTrustedURL('https://stackoverflow.com')).toEqual(false);
+  }));
+
+  it('should trust current URLs with same hostname as current one', inject([AppAuthService], (service: AppAuthService) => {
+   expect(service.isTrustedURL('https://api-staging.dd-decaf.eu/iam')).toEqual(true);
+   expect(service.isTrustedURL('https://api-staging.dd-decaf.eu/iam')).toEqual(true);
+ }));
+
+ it('should trust URLs which have been explicitly added', inject([AppAuthService], (service: AppAuthService) => {
+   service.trustedURLs.add('https://stackoverflow.com/questions');
```

Comment:
I guess we won't need this feature, `service.trustedURLs` can be a list taken from `environment.trustedURLs`